From 2965738f4ebb284720345c1912ced99ed5a153e5 Mon Sep 17 00:00:00 2001
From: Nicola Fontana <ntd@entidi.it>
Date: Sun, 15 Mar 2015 08:21:40 +0100
Subject: [PATCH 03/10] Use $(PKG_CONFIG) instead of pkg-config

On cross-compilation the pkg-config binary is usually substituted by a
wrapper (e.g. i686-w64-mingw32-pkg-config) that provides the proper
environment to the original pkg-config tool. Ensure to use this wrapper
when required: it is set in the PKG_CONFIG variable by configure.

On straight build PKG_CONFIG is set to pkg-config, so backward
compatibility is retained.
---
 Makefile-gir.am | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/Makefile-gir.am b/Makefile-gir.am
index 52f7ee3..6889c57 100644
--- a/Makefile-gir.am
+++ b/Makefile-gir.am
@@ -55,8 +55,8 @@ else
 endif
 
 # glib
-GLIB_INCLUDEDIR=$(shell pkg-config --variable=includedir glib-2.0)/glib-2.0
-GLIB_LIBDIR=$(shell pkg-config --variable=libdir glib-2.0)
+GLIB_INCLUDEDIR=$(shell $(PKG_CONFIG) --variable=includedir glib-2.0)/glib-2.0
+GLIB_LIBDIR=$(shell $(PKG_CONFIG) --variable=libdir glib-2.0)
 
 GLIB_LIBRARY=glib-2.0
 
@@ -92,8 +92,8 @@ GLib-2.0.gir: g-ir-scanner g-ir-compiler$(EXEEXT)
 gir/DBusGLib-1.0.typelib: GObject-2.0.gir
 
 # gobject
-GOBJECT_INCLUDEDIR=$(shell pkg-config --variable=includedir gobject-2.0)/glib-2.0
-GOBJECT_LIBDIR=$(shell pkg-config --variable=libdir gobject-2.0)
+GOBJECT_INCLUDEDIR=$(shell $(PKG_CONFIG) --variable=includedir gobject-2.0)/glib-2.0
+GOBJECT_LIBDIR=$(shell $(PKG_CONFIG) --variable=libdir gobject-2.0)
 
 GOBJECT_LIBRARY=gobject-2.0
 
@@ -120,8 +120,8 @@ GObject_2_0_gir_FILES = \
 BUILT_GIRSOURCES += GObject-2.0.gir
 
 # gmodule
-GMODULE_INCLUDEDIR=$(shell pkg-config --variable=includedir gmodule-2.0)/glib-2.0
-GMODULE_LIBDIR=$(shell pkg-config --variable=libdir gmodule-2.0)
+GMODULE_INCLUDEDIR=$(shell $(PKG_CONFIG) --variable=includedir gmodule-2.0)/glib-2.0
+GMODULE_LIBDIR=$(shell $(PKG_CONFIG) --variable=libdir gmodule-2.0)
 
 GMODULE_LIBRARY=gmodule-2.0
 
@@ -146,13 +146,13 @@ GModule_2_0_gir_FILES = $(GLIB_INCLUDEDIR)/gmodule.h \
 BUILT_GIRSOURCES += GModule-2.0.gir
 
 # gio
-GIO_INCLUDEDIR=$(shell pkg-config --variable=includedir gio-2.0)/glib-2.0
-GIO_LIBDIR=$(shell pkg-config --variable=libdir gio-2.0)
+GIO_INCLUDEDIR=$(shell $(PKG_CONFIG) --variable=includedir gio-2.0)/glib-2.0
+GIO_LIBDIR=$(shell $(PKG_CONFIG) --variable=libdir gio-2.0)
 
 GIO_LIBRARY=gio-2.0
 
 if HAVE_GIO_UNIX
-GIO_UNIX_HDRS=$(shell pkg-config --variable=includedir gio-unix-2.0)/gio-unix-2.0/gio/*.h
+GIO_UNIX_HDRS=$(shell $(PKG_CONFIG) --variable=includedir gio-unix-2.0)/gio-unix-2.0/gio/*.h
 GIO_UNIX_PACKAGES = gio-unix-2.0
 else
 GIO_UNIX_HDRS=
-- 
2.3.3

